src/wng_building/station_river_creator.py

```diff
@@ -1,3 +1,124 @@
+import copy
+
+from src.data_handling.data_handler import DataHandler
+
+
 class StationRiverCreator:
-    def __init__(self):
-        pass
+    """
+    Class for creating the following data structures: stations, rivers, completed rivers
+    """
+    def __init__(self, data_handler: DataHandler):
+        """
+        Constructor.
+        :param DataHandler data_handler: a DataHandler instance
+        """
+        self.data_handler = data_handler
+
+        self.stations = None
+        self.rivers = None
+        self.completed_rivers = None
+
+    def run(self) -> None:
+        """
+        Run function. Gets stations, rivers and completed rivers.
+        """
+        self.data_handler.run()
+
+        self.stations = self.create_stations()
+
+        self.rivers = self.create_rivers()
+
+        self.completed_rivers = self.create_completed_rivers()
+
+    def create_stations(self) -> dict:
+        """
+        Function for creating the stations data structure.
+        stations is a dictionary with reg-number keys and dictionary values like
+        2275: {
+            'river_name': Tisza,
+            'station_name': 'Szeged',
+            'EOVy': 735218.1,
+            'EOVx': 101317.2,
+            'null_point': 73.70
+        }
+        :return dict: dictionary of all stations
+        """
+        stations = {}
+        for reg_num in list(self.data_handler.reg_station_mapping.keys()):
+            station_name = self.data_handler.reg_station_mapping[reg_num]
+            river_name = self.data_handler.station_river_mapping[station_name]
+
+            stations[reg_num] = {
+                'river_name': river_name,
+                'station_name': station_name,
+                'EOVy': self.data_handler.station_coordinates[reg_num]['EOVy'],
+                'EOVx': self.data_handler.station_coordinates[reg_num]['EOVx'],
+                'null_point': self.data_handler.station_coordinates[reg_num]['null_point']
+            }
+
+        return stations
+
+    def create_rivers(self) -> dict:
```

Comment:
Ez menthet közvetlenül a tagváltozóba, ha később nem akarjuk kívül is meghívni. Így csak a hívás lesz a `run` metódusban.